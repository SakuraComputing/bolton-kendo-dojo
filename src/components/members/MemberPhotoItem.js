import React from 'react';

const MemberPhotoItem = (props) => {
    return (
        <div>
            <div className="image-frame" >
                <img className="image-item" src={`uploads/${props.filename}`} alt="photoU" />
                    {props.description}
                    <button onClick={props.onUploadDelete} className="btn-delete">Delete Photo</button>
            </div>
        </div>
    );
}

export default MemberPhotoItem;
