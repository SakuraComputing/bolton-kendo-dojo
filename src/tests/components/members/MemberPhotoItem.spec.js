import React from 'react'
import { shallow } from 'enzyme';
import MemberPhotoItem from '../../../components/members/MemberPhotoItem';

describe('Member Photo Item', () => {
    it('should render correctly', () => {
        const onUploadDelete = jest.fn();
        const photo = shallow(<MemberPhotoItem onUploadDelete={onUploadDelete}/>)
        expect(photo).toMatchSnapshot();
    });
});