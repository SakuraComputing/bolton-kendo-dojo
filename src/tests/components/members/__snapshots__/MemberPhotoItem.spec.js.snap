// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Photo Item should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemberPhotoItem
    onUploadDelete={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="image-frame"
      >
        <img
          alt="photoU"
          className="image-item"
          src="uploads/undefined"
        />
        <button
          className="btn-delete"
          onClick={[MockFunction]}
        >
          Delete Photo
        </button>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="photoU"
            className="image-item"
            src="uploads/undefined"
          />,
          undefined,
          <button
            className="btn-delete"
            onClick={[MockFunction]}
          >
            Delete Photo
          </button>,
        ],
        "className": "image-frame",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "photoU",
            "className": "image-item",
            "src": "uploads/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete Photo",
            "className": "btn-delete",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Delete Photo",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="image-frame"
        >
          <img
            alt="photoU"
            className="image-item"
            src="uploads/undefined"
          />
          <button
            className="btn-delete"
            onClick={[MockFunction]}
          >
            Delete Photo
          </button>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="photoU"
              className="image-item"
              src="uploads/undefined"
            />,
            undefined,
            <button
              className="btn-delete"
              onClick={[MockFunction]}
            >
              Delete Photo
            </button>,
          ],
          "className": "image-frame",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "photoU",
              "className": "image-item",
              "src": "uploads/undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete Photo",
              "className": "btn-delete",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Delete Photo",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
